src/common/debug/Log.h

```diff
@@ -43,44 +43,19 @@ class MockLogger {
 public:
   /// \brief Mocked log method that can be mocked to check if the logger is
   /// called with the correct parameters.
-  MAKE_MOCK2(log,
-             void(const std::string &category, const std::string &message));
+  MOCK_METHOD(void, log, (const std::string &category, const std::string &message), ());
+
+  MockLogger() { instance = this; };
+  virtual ~MockLogger() { instance = nullptr; };
```

Comment:
You can skip the `;` when you provide the implementation

```suggestion
  MockLogger() { instance = this; }
  virtual ~MockLogger() { instance = nullptr; }
```